src/dmsort.rs

```diff
@@ -205,7 +205,7 @@ impl<'a, T> Drop for DmSorter<'a, T> {
 			// This code will only run on stack-unwind (panic).
 
 			// Move back all elements into the slice:
-			ptr::copy_nonoverlapping(self.dropped.as_ptr(), &mut self.slice[self.write], self.dropped.len());
+			ptr::copy_nonoverlapping(self.dropped.as_ptr(), self.slice.as_mut_ptr().add(self.write), self.dropped.len());
```

Comment:
In this case, the pointer created here is a pointer to a single element, but this call actually needs a pointer which has provenance over a range. This could be created with something like `&mut self.slice[self.write..][..self.dropped.len()] as *mut T` but just offsetting a pointer to the whole slice is simpler.